service/history/fx.go

```diff
@@ -170,6 +173,31 @@ func TelemetryInterceptorProvider(
 	)
 }
 
+func HealthSignalAggregatorProvider(
+	dynamicCollection *dynamicconfig.Collection,
+	metricsHandler metrics.Handler,
+	logger log.ThrottledLogger,
+) interceptor.HealthSignalAggregator {
+	return interceptor.NewHealthSignalAggregatorImpl(
+		dynamicconfig.PersistenceHealthSignalWindowSize.Get(dynamicCollection)(),
+		dynamicconfig.PersistenceHealthSignalBufferSize.Get(dynamicCollection)(),
```

Comment:
could it use its own config?

Reply:
We could, but I don't want to overcomplicate before we feel it's necessary! LMK if you have strong opinions